rasa/core/lock_store.py

```diff
@@ -223,19 +226,32 @@ def __init__(
             ssl=use_ssl,
             socket_timeout=socket_timeout,
         )
+
+        if (prefix is None) or (prefix == ""):
+            self.prefix = "lock:"
+        elif (prefix is not None) and (
+            (isinstance(prefix, str)) and (prefix.isalnum())
+        ):
+            self.prefix = prefix + ":lock:"
```

Comment:
why have `prefix:lock:` instead of just `prefix:` if one is provided? the same goes for the tracker store. I would expect `<prefix>:` to _replace_ the default `lock:` one

Reply:
same here, a debug message would be helpful

Reply:
If we only use `<prefix>:` and it happens to be the same prefix as used in both the lock store and tracker stores then they will collide. If the keys are`<prefix>:lock:` and `<prefix>:tracker:` then they can work in the same redis db without conflict.

Reply:
okay, ideally we should compare the prefixes provided for the lock store and the tracker store then. but I think this solution works as well. we should definitely omit a debug message indicating what the prefix is actually set to